[libexif] Build failure on arm64-osx

Package: libexif:arm64-osx -> 0.6.24#1

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 15.0.0.15000040
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: 72010900b 2023-12-04 (7 days ago)

**To Reproduce**

`vcpkg x-set-installed --enforce-port-checks --allow-unsupported libexif[core]`
**Failure logs**

```
-- Using cached libexif-libexif-v0.6.24.tar.gz.
-- Extracting source /Users/leanderSchulten/git_projekte/vcpkg/downloads/libexif-libexif-v0.6.24.tar.gz
-- Applying patch fix-ssize.patch
-- Using source at /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/src/v0.6.24-b5073adc7f.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for arm64-osx-dbg
-- Getting CMake variables for arm64-osx-rel
-- Generating configure for arm64-osx
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /opt/homebrew/bin/autoreconf -vfi
    Working Directory: /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/src/v0.6.24-b5073adc7f.clean/
    Error code: 1
    See logs for more information:
      /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/autoconf-arm64-osx-out.log
      /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/autoconf-arm64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:629 (vcpkg_execute_required_process)
  ports/libexif/portfile.cmake:20 (vcpkg_configure_make)
  scripts/ports.cmake:170 (include)



```
<details><summary>/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/autoconf-arm64-osx-out.log</summary>

```
glibtoolize: putting auxiliary files in '.'.
glibtoolize: copying file './ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'auto-m4'.
glibtoolize: copying file 'auto-m4/libtool.m4'
glibtoolize: copying file 'auto-m4/ltoptions.m4'
glibtoolize: copying file 'auto-m4/ltsugar.m4'
glibtoolize: copying file 'auto-m4/ltversion.m4'
glibtoolize: copying file 'auto-m4/lt~obsolete.m4'
```
</details>

<details><summary>/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/libexif/autoconf-arm64-osx-err.log</summary>

```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: true --force
autoreconf: running: aclocal --force -I auto-m4 -I m4m
autoreconf: configure.ac: tracing
autoreconf: running: glibtoolize --copy --force
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I auto-m4 -I m4m
autoreconf: running: /opt/homebrew/Cellar/autoconf/2.71/bin/autoconf --force
autoreconf: running: /opt/homebrew/Cellar/autoconf/2.71/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:73: installing './ar-lib'
configure.ac:72: installing './compile'
configure.ac:74: installing './config.guess'
configure.ac:205: error: required file './config.rpath' not found
configure.ac:74: installing './config.sub'
configure.ac:10: installing './install-sh'
configure.ac:10: installing './missing'
Makefile.am: installing './INSTALL'
configure.ac:205: error: required file './ABOUT-NLS' not found
contrib/examples/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
autoreconf: error: automake failed with exit status: 1
```
</details>

--- a/ports/libexif/portfile.cmake
+++ b/ports/libexif/portfile.cmake
@@ -1,4 +1,3 @@
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libexif/libexif
@@ -13,7 +12,6 @@ vcpkg_list(SET options)
 if("nls" IN_LIST FEATURES)
     vcpkg_list(APPEND options "--enable-nls")
 else()
-    set(ENV{AUTOPOINT} true) # true, the program
     vcpkg_list(APPEND options "--disable-nls")
 endif()
 

--- a/ports/libexif/vcpkg.json
+++ b/ports/libexif/vcpkg.json
@@ -1,19 +1,21 @@
 {
   "name": "libexif",
   "version": "0.6.24",
-  "port-version": 1,
+  "port-version": 2,
   "description": "a library for parsing, editing, and saving EXIF data",
   "homepage": "https://libexif.github.io/",
   "license": "LGPL-2.1-or-later",
   "supports": "!uwp",
   "dependencies": [
     {
+      "$comment": "Needs gettext m4 files and autopoint during autoreconf. No runtime dependency.",
+      "name": "gettext",
+      "host": true
+    },
+    {
       "name": "vcpkg-cmake",
       "host": true
     }
-  ],
-  "default-features": [
-    "nls"
   ],
   "features": {
     "nls": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4270,7 +4270,7 @@
     },
     "libexif": {
       "baseline": "0.6.24",
-      "port-version": 1
+      "port-version": 2
     },
     "libfabric": {
       "baseline": "1.13.2",

--- a/versions/l-/libexif.json
+++ b/versions/l-/libexif.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "de3b3a5ade33ccafbeb4a00cde07954717240eee",
+      "version": "0.6.24",
+      "port-version": 2
+    },
+    {
       "git-tree": "40f7058e18a937bab35b7213595fc4eef5de7dec",
       "version": "0.6.24",
       "port-version": 1

